src/Service/Auth/auth.service.ts

```diff
@@ -0,0 +1,120 @@
+import { Request, Response, NextFunction } from "express";
+import { StatusCodes } from "http-status-codes";
+import { UserType } from "../../Models/user.model";
+import AuthRepository from "../../Repository/auth.repository";
+import validation from "../../Utils/ValidationUtil/validation"
+import {
+    BadRequestError, 
+    NotFoundError, 
+    UnAuthenticatedError
+} from "../../Utils/ErrorUtil"
+import bcrypt from "../../Utils/bcrypt.encrypt"
+
+
+// instances
+const authRepository = new AuthRepository;
+const validations = new validation;
+
+
+export default class AuthService {
```

Comment:
How come your service has all the access to Request and Response but your controllers are not doing anything?
For instance **const { userName, password, country, phoneNumber, email } = req.body;** Should be done inside a controller and passes down to service

Reply:
Oh. I will make necessaries corrections and make changes